Clarify docs around disk capacity expectation. (#115745)
elastic/elasticsearch: docs/reference/datatiers.asciidoc

For example hardware profiles, refer to Elastic Cloud's {cloud}/ec-reference-hardware.html[instance configurations]. 
Review our recommendations for optimizing your storage for <<indexing-use-faster-hardware,indexing>> and <<search-use-faster-hardware,search>>.

> IMPORTANT: {es} generally expects nodes within a data tier to share the same hardware profile (such as CPU, RAM, disk capacity) 

Reviewer: ```suggestion
IMPORTANT: By default, {es} assumes nodes within a data tier share the same hardware profile (such as CPU, RAM, disk capacity) 
```
Should we potentially take out "generally" and/or add "by default" in order to firm this as baseline expectation? Or go as far to say ES "assumes" this is true so if you want to do different you need to "carefully architect" (I like that phrasing)

Reviewer 2: I'd just say "Elasticsearch expects"

> as it spreads the load equally across all nodes within the data tier.
> Data tiers with unequally resourced nodes are at risk of <<hotspotting,hot spotting>>.

Reviewer 3: I do not think it is true today that we spread the load equally. I also think it is not a guarantee to avoid hotspotting, which is sort of implied here.

```suggestion
.
Data tiers with unequally resourced nodes have a higher risk of <<hotspotting,hot spotting>>.
```

Reviewer 4: ++ although we do not spread the load equally, we allocate the load _assuming_ that the available resources are equal.

> Variations not following this recommendation should be carefully architected to avoid <<hotspotting,hot spotting>>.

Reviewer 2: consider

```suggestion
Data tiers with unequally resourced nodes are at risk of <<hotspotting,hot spotting>>.
```

Reviewer: I agree with your point. 

I believe the phrasing is intended to push the customer towards, while you may get it to work, Elastic doesn't "support" figuring this out for you. If we leave it at "nodes are at risk", we're going to keep getting supportCases+devEscalations which I believe is why we're updating this to say "you did it to yourself" by going outside Elasticsearch assumptions. Could we workshop a middle ground phrasing?

Reviewer 2: how about

```suggestion
Data tiers with unequally resourced nodes are at risk of <<hotspotting,hot spotting>>. These configurations are not supported.
```

I think the current phrasing will lead to people looking for guidance of what "careful" could look like

> Variations not following this recommendation should be carefully architected to avoid <<hotspotting,hot spotting>>.

Reviewer 2: This construction is a little confusing. What steps could someone take to "carefully architect" a node to avoid hot spotting? if we can't provide recommendations, I'd change this to "Data tiers with unequally resourced nodes are at risk of hot spotting."
